### PR TITLE
Ported BERT pre-training tutorial

### PR DESCRIPTION
**Pull request by ShadenSmith**
You can view the tutorial here: http://shaden.io/ghpages-test/tutorials/bert-pretraining/

### CODE REVIEW

File: docs/_tutorials/bert-pretraining.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+We will go through how to setup the data pipeline and how to run the original
+BERT model. Then we will show step-by-step how to modify the model to
+leverage DeepSpeed. Finally, we demonstrate the performance evaluation and
+memory usage reduction from using DeepSpeed.
+
+## Pre-training Bing BERT without DeepSpeed
+
+We work from a adaptations of
+[huggingface/transformers](https://github.com/huggingface/transformers) and
+NVIDIA. We have forked this repo under
```
Comment (jeffra) [inline docs/_tutorials/bert-pretraining.md:22]: I'd change this to "We work from adaptations of ...", removing 'a'.

Also I'd add a link for nvidia to this repo: https://github.com/NVIDIA/DeepLearningExamples

File: docs/_tutorials/bert-pretraining.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+[DeepSpeedExamples/bing_bert](https://github.com/microsoft/DeepSpeedExamples/tree/master/bing_bert)
+and made several modifications in their script:
+  * We adopted the modeling code from NVIDIA's BERT under `bing_bert/nvidia/`.
+  * We extended the data pipeline from [Project Turing](https://msturing.org/)
+    under `bing_bert/turing/`.
+
+Before using DeepSpeed, it is recommended to first go over the pre-requisites
+from the original model README, such as the overview of the model's structure
+and how to run the model.
```
Comment (jeffra) [inline docs/_tutorials/bert-pretraining.md:31]: Not sure what to do about this, maybe remove it? Our bing_bert example in our repo doesn't have a readme currently, however it probably should :( Maybe we can add a readme there that just links to this tutorial once it's live?

File: docs/_tutorials/bert-pretraining.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+### Training Data Setup
+
+Download the Wikipedia and BookCorpus datasets and specify their paths in the
+model config file `DeepSpeedExamples/bing_bert/bert\_large\_adam\_seq128.json`:
+
```
Comment (jeffra) [inline docs/_tutorials/bert-pretraining.md:37]: Can we add text saying that download/preprocessing instructions are coming soon?

File: docs/_tutorials/bert-pretraining.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+| Parameters               | 128 Sequence              | 512 Sequence              |
+| ------------------------ | ------------------------- | ------------------------- |
+| Total batch size         | 16K                       | 16K                       |
+| Train micro batch size per gpu | 64                  | 8                         |
+| Optimizer                | Lamb                      | Lamb                      |
+| Learning rate            | 4e-3                      | 1e-3                      |
+| Min Lamb coefficient     | 0.08                      | 0.08                      |
+| Max Lamb coefficient     | 0.5                       | 0.5                       |
+| Learning rate scheduler  | `warmup_linear_decay_exp` | `warmup_linear_decay_exp` |
+| Warmup ratio             | 0.02                      | 0.01                      |
```
Comment (jeffra) [inline docs/_tutorials/bert-pretraining.md:297]: In the code itself the 'warmup ratio' parameter is actually set with 'warmup_proportion'

File: docs/_tutorials/bert-pretraining.md (hunk 2 above)
Comment (ShadenSmith) [reply docs/_tutorials/bert-pretraining.md:31]: Good point. I'll just remove from now to simplify things and avoid sending folks in a circle. We should absolutely go through the examples repo and add readmes, etc. Will save that for a future PR though.

File: docs/_tutorials/bert-pretraining.md (hunk 3 above)
Comment (ShadenSmith) [reply docs/_tutorials/bert-pretraining.md:37]: Done

File: docs/_tutorials/bert-pretraining.md (hunk 1 above)
Comment (ShadenSmith) [reply docs/_tutorials/bert-pretraining.md:22]: Done

File: docs/_tutorials/bert-pretraining.md (hunk 4 above)
Comment (ShadenSmith) [reply docs/_tutorials/bert-pretraining.md:297]: Fixed
